doc/source/whatsnew/v0.21.1.txt

```diff
@@ -73,7 +73,7 @@ Indexing
 - Bug in a boolean comparison of a ``datetime.datetime`` and a ``datetime64[ns]`` dtype Series (:issue:`17965`)
 - Bug where a ``MultiIndex`` with more than a million records was not raising ``AttributeError`` when trying to access a missing attribute (:issue:`18165`)
 - Bug in :class:`IntervalIndex` constructor when a list of intervals is passed with non-default ``closed`` (:issue:`18334`)
--
+- Bug in ``IntervalIndex.symmetric_difference()`` where the symmetric difference with a non-``IntervalIndex`` did not raise (:issue:`18475`)
```

Comment:
move this to 0.22. I am not sure we can backport the actual test changes w/o causing some hassle.

Alternatively you can split this to 2 pr's, 1 the test changes and 1 the perf fix (which we can backport).

Reply:
Moved to 0.22.0